llvm/lib/Support/Unix/Memory.inc

```diff
@@ -177,6 +181,78 @@ std::error_code Memory::protectMappedMemory(const MemoryBlock &M,
       alignAddr((const uint8_t *)M.Address + M.AllocatedSize, PageSize);
 
   bool InvalidateCache = (Flags & MF_EXEC);
+  bool SkipMprotect = false;
+
+#if defined(__linux__)
```

Comment:
Could this all be sunk into a helper function?
```
#if defined(__linux__)
whatever = do_a_bunch_of_stuff();
#endif
```

Reply:
It's much less code in the current version and moving it to a helper function would require passing quite some arguments. So, maybe, it's okay now?

I could move the memfd part to a helper class to squeeze out a few more lines, but that feels like overengineering just for the cause.